src/ui/features/onboarding/Success/WebAppMessageHandler.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useRef } from 'react';
+import { invariant } from 'src/shared/invariant';
+import { isObj } from 'src/shared/isObj';
+
+const ZERION_WEB_APP_URL = new URL(
+  process.env.NODE_ENV === 'development'
+    ? 'http://localhost:3000'
+    : 'https://app.zerion.io'
+);
+
+type WebAppCallbackMethod = 'set-referral-code';
+
+interface WebAppMessage<T = unknown> {
```

Comment:
We don't use generic value of this interface anywhere. But this is probably not a problem.

Reply:
Let's just have `params?: unknown` for now ✅